test/1_TokenVesting.js

```diff
@@ -0,0 +1,64 @@
+var TokenVesting = artifacts.require('TokenVesting');
+const MintableToken = artifacts.require('MintableToken');
+contract("token vesting",function(accounts){
+    var token;
+    var tokenVest;
+ it("should be approve token vesting contracts the tokens",function(){
+     
+   return MintableToken.deployed().then(function(instance){
+       token=instance;
+   return instance.mint(accounts[0],1000,{from:accounts[0]});
+   }).then(function(res){
+       return token.totalSupply.call();
+   }).then(function(supp){
+      return TokenVesting.deployed()
+   }).then(instance2=>{
+       tokenVest=instance2;
+       return token.approve(tokenVest.address,100,{from:accounts[0]})
+   }).catch(err=>{
+       assert.isUndefined(err,"no error");
+   })
+})
+it("should not be able to claim funds before vesting",function(){
+    return tokenVest.claimTokens().then(res=>{
+
+    }).catch(err=>{
+        assert.isDefined(err,"error");
+    });
+})
+it("should be able to fund vesting",function(){
+  return tokenVest.fundVesting(100).then((res)=>{
+  }).then(function(res){
+    assert.isUndefined(res,"no error");
+  });
+})
+it("should be able to fund vesting when contract is not approved for such amount of tokens",function(){
+    return tokenVest.fundVesting(1000).then((res)=>{
+    }).catch(function(err){
+      assert.isDefined(err,"there is err");
+    });
+  })
+it("should be not able to fund vesting again after doing it once",function(){
+    return tokenVest.fundVesting(100).then((res)=>{
+    }).catch(function(err){
+      assert.isDefined(err,err);
+    });
+  })
+    
+it("should  not be able to claim funds before the cliff",function(){
+    return tokenVest.claimTokens().then(res=>{
+
+    }).catch(err=>{
+        assert.isDefined(err,"no error");
+    });
+})
+it("owner should be able to kill vesting",function(){
```

Comment:
Where are any tests checking that vesting actually works and sum releassed is as expected?

Reply:
Yes, seems that those tests are missing.

Reply:
ping @kpulkit29